### PR TITLE
Random POVMs, plus random isometries

### PR DESCRIPTION
**Pull request by lpawela**

### CODE REVIEW

**Comment by lpawela** [comment]
This resolves #49

File: randommatrices/src/circular.jl [hunk 1]
```diff
@@ -40,10 +41,8 @@ end
 function rand(c::CSE)
     z = rand(c.g)
     u = _qr_fix(z)
-    #TODO this does not require matrix multiplication
-    a = diagm(-ones(c.d-1), 1) + diagm(ones(c.d-1), -1)
-    ur = -a*transpose(u)*a
-    ur*u
+    ur = cat([[0 -1; 1 0] for _=1:round(Int, c.d/2)]..., dims=[1,2])
```
Comment (pgawron) [inline randommatrices/src/circular.jl]: round(Int, c.d/2)] -> c.d \div 2

File: randommatrices/src/circular.jl [hunk 2]
```diff
@@ -75,3 +74,21 @@ function rand(c::CircularQuaternionEnsemble)
...
+    function HaarIsometry(idim::Int, odim::Int)
+        idim <= odim || throw(ArgumentError("idim can't be greater than odim"))
+        g = GinibreEnsemble{2}(odim, idim)
+        new(idim, odim, g)
+    end
+end
+
+function rand(c::HaarIsometry)
+    z = rand(c.g)
+    _qr_fix(z)
```
Comment (pgawron) [inline randommatrices/src/circular.jl]: _qr_fix() -> _qr_fix!() -- function modifies its argument

**Comment by lpawela** [comment]
04429e8b9cbf82fea4db2e07eb6e8c5d7588acb6 also relsoves #63

File: src/randomqobjects.jl [hunk 3]
```diff
@@ -76,37 +67,73 @@ function ChoiJamiolkowskiMatrices(d::Int)
...
-# """
-# $(SIGNATURES)
-# - `d`: dimension.
-#
-# Generates random Jamiolkowski state of dimension `d`.
-# """
+
+# Random POVMs implemented according to
+# https://arxiv.org/pdf/1902.04751.pdf
+abstract type AbstractHaarPOVM <: ContinuousMatrixDistribution
```
Comment (pgawron) [inline src/randomqobjects.jl]: `ContinuousMatrixDistribution` type hijacking!

**Pull request merged**